TileRarities.lua

```diff
@@ -0,0 +1,131 @@
+--[[
+    The data for the different chances and multiplier values can be found on Data:Tile_Chances
```

Comment:
I think how you have the rarities formatted atm is fine, I probably wouldn't change it. you could consider just formatting it like `rankOrder: ["Basic", "Plus", "Super"]` in the json and not have a difference between display & key in this data, but if you are concerned there might be other important properties in the future this may just complicate things for future editors

Reply:
Only concern I have is to handle the capitalization of the text. Tho I assume Lua has a `tolowercase` method or similar like pretty much any other coding language out there...

Reply:
Idk why you ever need lowercase but yes Lua can convert to lowercase

Reply:
I use lowercased keys with underscores in JSON instead of lower-camelcased ones, as it feels more readable for me, so having `ranks` not be lowercased could be problematic here as I use these names to also retrieve the data from the different chances entries.

As an example:
```json
{
	"ranks": [
		"Basic",
		"Plus",
		"Super"
	],
	"chances": [
		{
			"rarity": "Common",
			"basic": {
				"chance": 44.9,
				"tile_multiplier": 1,
				"gold_multiplier": 1,
				"wed_chance": 9,
				"sun_tile_multiplier": 2,
				"sun_gold_multiplier": 2
			}
        }
    ]
}
```
Here, values are stored under `basic` which - unless lua ignores casing here - wouldn't be accessible under `chances[].Basic`.

Anyways, `string.lower` does work for me (or there are at least no errors reported with it).